tasks/libs/common/gomodules.py

```diff
@@ -0,0 +1,323 @@
+from collections.abc import Callable
+from dataclasses import dataclass
+from functools import lru_cache
+from glob import glob
+from pathlib import Path
+from typing import ClassVar
+
+import yaml
+
+
+class GoModuleDumper(yaml.SafeDumper):
+    """SafeDumper that ignores aliases. (no references for readability)"""
+
+    def ignore_aliases(self, _):  # noqa
+        return True
+
+
+@dataclass
+class GoModule:
+    """A Go module abstraction.
+
+    See:
+        Documentation can be found in <docs/dev/modules.md>.
+
+    Args:
+        independent: specifies whether this modules is supposed to exist independently of the datadog-agent module. If True, a check will run to ensure this is true.
+
+    Usage:
+        A module is defined within a module.yml next to the go.mod file containing the following fields by default (these can be omitted if the default value is used):
+        > condition: always
+        > importable: true
+        > independent: true
+        > lint_targets:
+        > - .
+        > should_tag: true
+        > targets:
+        > - .
+        > used_by_otel: false
+
+        If a module should be ignored and not included within get_default_modules(), add the following to the module.yml:
+        > ignored: true
+    """
+
+    # Possible conditions for GoModule.condition
+    CONDITIONS: ClassVar[dict[str, Callable]] = {
+        'always': lambda: True,
+        'never': lambda: False,
+        'is_linux': lambda: sys.platform == "linux",
+    }
+
+    # Posix path of the module's directory
+    path: str
+    targets: list[str] | None = None
+    condition: str = 'always'
+    should_tag: bool = True
+    # HACK: Workaround for modules that can be tested, but not imported (eg. gohai), because
+    # they define a main package
+    # A better solution would be to automatically detect if a module contains a main package,
+    # at the cost of spending some time parsing the module.
+    importable: bool = True
+    independent: bool = True
+    lint_targets: list[str] | None = None
+    used_by_otel: bool = False
+    base_dir: Path | None = None
+
+    @staticmethod
+    def from_dict(path: str, data: dict[str, object], base_dir: Path | None = None) -> GoModule:
+        default = GoModule.get_default_attributes()
+
+        return GoModule(
+            path=path,
+            targets=data.get("targets", default["targets"]),
+            lint_targets=data.get("lint_targets", default["lint_targets"]),
+            condition=data.get("condition", default["condition"]),
+            should_tag=data.get("should_tag", default["should_tag"]),
+            importable=data.get("importable", default["importable"]),
+            independent=data.get("independent", default["independent"]),
+            used_by_otel=data.get("used_by_otel", default["used_by_otel"]),
+            base_dir=base_dir,
+        )
+
+    @staticmethod
+    def parse_path(dir_path: str | Path, base_dir: Path | None = None) -> tuple[str, Path, Path, Path]:
```

Comment:
I did this method because it is used before creating the module (to verify whether there is a `module.yml` file at the dedicated location). As it is used only on some methods, I think creating these attributes may pollute the dataclass fields. If you really think that it is better like this, tell me 👍